simtools/model/model_parameter.py

```diff
@@ -0,0 +1,561 @@
+        """
+        Get the value of an existing parameter of the model.
+
+        Parameters
+        ----------
+        par_name: str
+            Name of the parameter.
+        parameter_dict: dict
+            Dictionary with complete DB entry for the given parameter
+            (including the 'value', 'units' fields).
+
+        Returns
+        -------
+        Value of the parameter.
+
+        Raises
+        ------
+        InvalidModelParameter
+            If par_name does not match any parameter in this model.
+        """
+
+        parameter_dict = parameter_dict if parameter_dict else self.get_parameter(par_name)
+        try:
+            return parameter_dict.get("value")
+        except KeyError as exc:
+            self._logger.error(f"Parameter {par_name} does not have a value")
+            raise exc
+
+    def get_parameter_value_with_unit(self, par_name):
+        """
+        Get the value of an existing parameter of the model as an Astropy Quantity with its unit.\
+        If no unit is provided in the model, the value is returned without a unit.
+
+        Parameters
+        ----------
+        par_name: str
+            Name of the parameter.
+
+        Returns
+        -------
+        Astropy quantity with the value of the parameter multiplied by its unit. If no unit is \
+        provided in the model, the value is returned without a unit.
+
+        Raises
+        ------
+        InvalidModelParameter
+            If par_name does not match any parameter in this model.
+        """
+        _parameter = self.get_parameter(par_name)
+        _value = self.get_parameter_value(None, _parameter)
+        try:
+            _units = _parameter.get("unit") or _parameter.get("units")
+            return float(_value) * u.Unit(_units)
+        except (KeyError, TypeError):
+            return _value
+
+    def get_parameter_type(self, par_name):
+        """
+        Get the type of an existing parameter of the model.
+
+        Parameters
+        ----------
+        par_name: str
+            Name of the parameter.
+
+        Returns
+        -------
+        str or None
+            type of the parameter (None if no type is defined)
+
+        """
+        parameter_dict = self.get_parameter(par_name)
+        try:
+            return parameter_dict.get("type") or parameter_dict.get("Type")
+        except KeyError:
+            self._logger.debug(f"Parameter {par_name} does not have a type")
+        return None
+
+    def get_parameter_file_flag(self, par_name):
+        """
+        Get value of parameter file flag.
+
+        Parameters
+        ----------
+        par_name: str
+            Name of the parameter.
+
+        Returns
+        -------
+        bool
+            True if file flag is set.
+
+        """
+        parameter_dict = self.get_parameter(par_name)
+        try:
+            if parameter_dict.get("file") or parameter_dict.get("File"):
```

Comment:
This would require a renaming of all database entries. I suggest we do not do that and keep it as it is.

I've added one sentence to the docstring saying that this is a boolean.